src/day.c

```diff
@@ -734,26 +734,32 @@ int day_paste_item(struct day_item *p, time_t date)
 	return p->type;
 }
 
-/* Returns the position corresponding to a given item. */
-int day_get_position_by_aptev_ptr(union aptev_ptr aptevp)
+/*
+ * Returns the position of a day_item in the day_items vector.
+ * An item (event/appointment) may have several occurrences as day_item,
+ * but at most one on any day. This is found by (order, item) or by
+ * item alone.
+ */
+int day_get_position(time_t order, union aptev_ptr item)
 {
-	int n = 0;
+	int i = 0;
+	struct day_item *p;
 
-	VECTOR_FOREACH(&day_items, n) {
-		struct day_item *p = VECTOR_NTH(&day_items, n);
-		/* Compare pointers. */
-		if (p->item.ev == aptevp.ev)
-			return n;
+	VECTOR_FOREACH(&day_items, i) {
+		p = VECTOR_NTH(&day_items, i);
+		/* Same day and item? */
+		if (p->order == order && p->item.apt == item.apt)
+			return i;
+	}
+	VECTOR_FOREACH(&day_items, i) {
+		p = VECTOR_NTH(&day_items, i);
+		/* Same item? */
+		if (p->item.apt == item.apt)
+			return i;
```

Comment:
I found this confusing and reading the commit message or the comments did not help. What is the purpose of the new `order` argument? What does this double search do? Without reading the remaining code in detail: I guess the first loop tries to find an occurrence on the current day? But then, what does the other loop do? What exactly is the (new?) purpose of this function?

Also, again I think we can skip the `Multiple days` from the commit message. It makes it overly long and does not add any value.

Reply:
Re `order`: see commit log message https://github.com/lfos/calcurse/pull/193/commits/be8a363208ab7fc09644c628e43255b6eeb167ed.

Reply:
> I found this confusing and reading the commit message or the comments did not help. What is the purpose of the new `order` argument? What does this double search do? Without reading the remaining code in detail: I guess the first loop tries to find an occurrence on the current day? But then, what does the other loop do? What exactly is the (new?) purpose of this function?

I'm not satisfied with this myself, and the comment is of the sort I try to avoid: saying what the code itself says.

The problem of setting the selected item correctly in the `day_items` vector (appointments list), when the vector/list is rebuilt, existed before. I changed the solution in commit https://github.com/lfos/calcurse/pull/193/commits/dd6a2a305091216a95b9cf65f303a929b0421d9b, but am no longer sure I like the way it was changed. I am going to look it over and update the PR as appropriate.

Reply:
I rewrote it partly. See 4a19b95 and the commit message which has a comparison with "single day" calcurse and explains the differences.